src/Autofac.Diagnostics.DotGraph/InstanceEqualityComparer.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Autofac.Diagnostics.DotGraph
+{
+    /// <summary>
+    /// Equality comparer that determines if two resolve requests are effectively the
+    /// same based on the returned instance. Used to find "duplicates" in the graph
+    /// during normalization.
+    /// </summary>
+    internal class InstanceEqualityComparer : IEqualityComparer<ResolveRequestNode>
```

Comment:
Proposed rename: `NodeEqualityComparer`. `InstanceEqualityComparer` is a bit generic.

Reply:
Sure. I had called it that because it compares the instance inside the node, but I see the confusion.

Reply:
`NodeEqualityComparer` it is!